frame/contracts/rpc/src/lib.rs

```diff
@@ -89,11 +92,22 @@ pub struct InstantiateRequest<AccountId, Hash> {
 	origin: AccountId,
 	endowment: NumberOrHex,
 	gas_limit: NumberOrHex,
+	storage_limit: Option<NumberOrHex>,
 	code: Code<Hash>,
 	data: Bytes,
 	salt: Bytes,
 }
 
+/// A struct that encodes RPC parameters required for a call to upload a new code.
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(deny_unknown_fields)]
+pub struct CodeUploadRequest<AccountId> {
+	origin: AccountId,
+	code: Bytes,
+	storage_limit: Option<NumberOrHex>,
```

Comment:
I wonder if we should name this `deposit_limit` instead. It's denoted as `Balance` and afaiu limits how much deposit can be charged from the caller. To me, the term `storage_limit` indicates some limit in terms of byte size.

Reply:
But `deposit_limit` you need to ask "deposit for what?". And using `storage_deposit_limit` is too long. Its quite the dilemma.

Reply:
Why is it too long?

Reply:
Fair enough. I renamed it.